server/public/index.php

```diff
@@ -1,15 +1,13 @@
 <?php
 
-use \Psr\Http\Message\ServerRequestInterface as Request;
-use \Psr\Http\Message\ResponseInterface as Response;
+require __DIR__ . '/../vendor/autoload.php';
 
-require __DIR__ . '/../../server/vendor/autoload.php';
+$settings = require __DIR__ . '/../src/settings.php';
 
-$app = new \Slim\App;
-$app->get('/hello[/{name}]', function (Request $request, Response $response, array $args) {
-	$name = $args['name'] ?: 'world';
-	$response->getBody()->write("Hello, $name");
+$app = new \Slim\App( $settings );
+
+require __DIR__ . '/../src/dependencies.php';
+
+require __DIR__ . '/../src/routes.php';
```

Comment:
Instead of adding this here, just autoload the files.
https://getcomposer.org/doc/04-schema.md#files

but you'd probably need to put `$app = new \Slim\App( $settings );` in it's own file so they can be autoloaded in order. 

Then again, I'm not sure if it will globally scope it or not. I suppose you could always make a class that would return a singleton.

But if you're doing that, might be best to just create a class that handles creating the dependencies and the routes. Then that class can just be namespaced and and autoloaded (psr-4) like normal and you can just call some `run()` method on your class.

Reply:
We need the includes in a specific order. I don't think that wrapping things in a class just so we can autoload some files brings any benefits. Not right now anyway. We don't have to run any complex logic to bootstrap the app so I think this should do it. We can change it down the line if we see we have to of course

Reply:
I mean the files autoloader will load it in order, but I think it prevents globally scoped variables.

Reply:
We need settings before we create the app instance, and we need the app instance before registering routes and dependencies. Afaik that's not possible with autoload config

Reply:
yeah the settings would have to be a function/method that would return the settings. and at that point it'd be cleaner to just wrap it in a class.

_meh_ don't worry about it. not a big deal.

Reply:
I mean, It definitely can be done. If you have a strong preference against how it is done now lets discuss it offline and change it, unless you want that change to be part of this.

Reply:
_meh_ not a big deal, but my preference is _towards_ autoloading (if possible) but seems a little annoying with Slim (or any micro framework for that matter)